### PR TITLE
libfreenect2 initial commit

### PR DESCRIPTION
**Pull request by ybott**
Support for Kinect version 2 depth camera.

### CODE REVIEW

File: Library/Formula/libfreenect2.rb [hunk 1]
```diff
@@ -0,0 +1,143 @@
+require "formula"
```
Comment (DomT4) [inline Library/Formula/libfreenect2.rb]: Hasn't been necessary for a while. Can kill it.

File: Library/Formula/libfreenect2.rb [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+class Libfreenect2 < Formula
+  desc "Drivers/Example for Kinect V2"
+  homepage "https://github.com/OpenKinect/libfreenect2"
+  url "https://github.com/OpenKinect/libfreenect2/archive/master.zip"
```
Comment (DomT4) [inline Library/Formula/libfreenect2.rb]: We'll need an actual release. The `master.zip` will update & break the checksum every time someone pushes a new commit there. Suggest you ask upstream if they're willing to tag a release for you here.

File: Library/Formula/libfreenect2.rb [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  version "0.0.1"
+  
+  head "https://github.com/OpenKinect/libfreenect2.git"
+
+  option :universal
+
+  depends_on "libusb"
+  depends_on "nasm" => :recommended
+  depends_on "jpeg-turbo" => :recommended
+  depends_on "homebrew/science/opencv" => :recommended
```
Comment (DomT4) [inline Library/Formula/libfreenect2.rb]: Can't have `:recommended` or mandatory cross-tap dependencies in the core. These would have to be `:optional` at the very most.

File: Library/Formula/libfreenect2.rb [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+  depends_on "homebrew/versions/glfw3" => :recommended
```
Comment (DomT4) [inline Library/Formula/libfreenect2.rb]: Can't have `:recommended` or mandatory cross-tap dependencies in the core. These would have to be `:optional` at the very most.

File: Library/Formula/libfreenect2.rb [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+  depends_on "pkg-config" => :build
+  depends_on "cmake" => :build
+
+  patch :DATA
+
+  def install
+    args = "-DCMAKE_INSTALL_PREFIX=/usr/local/Cellar/libfreenect2/#{version}"
```
Comment (DomT4) [inline Library/Formula/libfreenect2.rb]: We don't hardcore `/usr/local`. You can put Homebrew anywhere, `/usr/local` is just our preference and where our binaries are targeted. Use `system "cmake", "examples/protonect", *std_cmake_args` instead.

File: Library/Formula/libfreenect2.rb [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+  depends_on "libusb"
+  depends_on "nasm" => :recommended
+  depends_on "jpeg-turbo" => :recommended
+  depends_on "homebrew/science/opencv" => :recommended
+  depends_on "homebrew/versions/glfw3" => :recommended
+  depends_on "pkg-config" => :build
+  depends_on "cmake" => :build
+
+  patch :DATA
```
Comment (DomT4) [inline Library/Formula/libfreenect2.rb]: Please add an upstream link to the bug submission/commit containing this patch, and a short summary of why it is necessary.

File: Library/Formula/libfreenect2.rb [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+  depends_on "libusb"
+  depends_on "nasm" => :optional
+  depends_on "jpeg-turbo" => :optional
+  depends_on "homebrew/science/opencv" => :optional
+  depends_on "homebrew/versions/glfw3" => :optional
+  depends_on "pkg-config" => :build
+  depends_on "cmake" => :build
+
+  patch :DATA
```
Comment (bfontaine) [inline Library/Formula/libfreenect2.rb]: What is this patch for? Did you submit it upstream? Could you add a comment above this line explaining why it’s necessary? Thanks!

File: Library/Formula/libfreenect2.rb [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+
+  def install
+    system "cmake", "examples/protonect", *std_cmake_args
+    system "make"
+    system "make", "install"
+  end
+end
```
Comment (bfontaine) [inline Library/Formula/libfreenect2.rb]: Please add a `test do` block.

File: Library/Formula/libfreenect2.rb [hunk 9]
```diff
@@ -0,0 +1,128 @@
+class Libfreenect2 < Formula
+  desc "Drivers/Example for Kinect V2"
+  homepage "https://github.com/OpenKinect/libfreenect2"
+  url "https://github.com/ybott/libfreenect2/archive/v0.0.1-alpha.tar.gz"
```
Comment (DomT4) [inline Library/Formula/libfreenect2.rb]: This release *needs* to come from the upstream repository, not your own fork. Only an official upstream release is acceptable I'm afraid.

**Comment by dunn** [comment]
I'm assuming you're using your fork because upstream hasn't tagged a stable release?  In these cases we ask that formula be added to our head-only repository.  There's actually already a PR open there for this: Homebrew/homebrew-head-only#153.

If you think the developers will make a stable release soon, we can leave this open. Otherwise I think we'll close it in favor of the head-only submission.

**Pull request closed without merging**